dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -563,6 +532,34 @@
 			<_AreAnyAssembliesTrimmed Condition="'$(_AreAnyAssembliesTrimmed)' == '' And '$(TrimMode)' == 'copy' And '$(_AssembliesWithCustomTrimMode)' == '$(_AssembliesWithCopyTrimMode)'">false</_AreAnyAssembliesTrimmed>
 			<_AreAnyAssembliesTrimmed Condition="'$(_AreAnyAssembliesTrimmed)' == ''">true</_AreAnyAssembliesTrimmed>
 
+			<!--
+			Registrar configuration:
+				- Set 'managed-static' for:
+					- NativeAOT supported platforms in all configurations
+					- ios/tvos device builds in all configurations
+					- macOS/MacCatalyst builds in release configuration
+				- Set 'partial-static' for:
+					- when no assemblies are trimmed and when _MarshalManagedExceptionMode is default (or not set)
+				- Otherwise set 'dynamic' 
+			-->
+			<Registrar Condition="'$(Registrar)' == '' And '$(_UseNativeAot)' == 'true'">managed-static</Registrar>
+			<Registrar Condition="'$(Registrar)' == '' And '$(_SdkIsSimulator)' != 'true' And ('$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS')">managed-static</Registrar>
+			<Registrar Condition="'$(Registrar)' == '' And '$(_BundlerDebug)' != 'true' And ('$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst')">managed-static</Registrar>
```

Comment:
Even though we are selecting the same value I separated the lines so it is clear what are the conditions.
Maybe this can eventually become a one-liner, but I think it would be hard to read.